### DISCUSSION TITLE
MOAB building difficulties in blueCFD

### DISCUSSION THREAD

**Discussion by DaleKramer** [blueCFD/Core#226] [Show and tell]
I am having difficulties building MOAB in the blueCFD environment.

https://bitbucket.org/fathomteam/moab/src/master/

I do not seem to have access to 'autoreconf' which is what is stopping me.

Is it possible to achieve this build in blueCFD terminal?

**Comment by wyldckat** [comment]
blueCFD-Core's terminal is actually MSys2, which means that any packages you may need, can be gotten by following their instructions. For example, see this report: https://github.com/msys2/MINGW-packages/issues/1010

Please also see issue #129 here at the blueCFD-Core project, because there are instructions on how to update your installation of MSys2 and how to support both an old installation of MSys2 and a new installation of MSys2 for the same blueCFD-Core installation.

We have plans to make a release of blueCFD-Core 2020-1 sometime this year, which will have an up-to-date installation of MSys2, but we don't have a release date yet.


Beyond this, I have never tried to build MOAB myself, so I don't know if it will work. You might want to do a clean installation of MSys2 separately and test building MOAB there, in case you want to avoid having to reinstall blueCFD-Core between attempts.

**Comment by DaleKramer** [comment]
Thank-you for the reply.  I am unable to follow the instructions at https://github.com/msys2/MINGW-packages/issues/1010 since 'pacman -S base-devel' gives this error in my blueCFD command line (and I have not been able to track down its meaning):
![image](https://user-images.githubusercontent.com/7749699/74384371-1a169000-4dbf-11ea-973e-8935f578f20a.png)

Also, you can see I tried a few fixes for database locking that I found on the internet.

Is this error due to the msys2 version used in a blueCFD-Core 2017-2 install ❓

**Comment by wyldckat** [comment]
Sorry, I forgot to explain about that ahead of time: the default installation options of blueCFD-Core will install in `C:\Program Files\blueCFD-Core-2017`, which only has partial permissions for us to write data on those folders. Which means that the MSys2 installation is frozen, due to security reasons (reducing the chances of there being a virus infection on those binaries).

One solution, even if annoying, is to uninstall blueCFD-Core from the current location and install it at a convenient location, for example: `C:\blueCFD`
Another solution, which might be simpler, but takes some time as well, is to:

1. Right click on the folder `C:\Program Files\blueCFD-Core-2017` and go to the Properties option at the bottom of the menu.
2. Go to the `Security` tab and edit the folder's permissions in a way that allows all users to read-write on that folder and sub-folders.
3. It takes 1-2 minutes or more for the permissions to propagate to all sub-folders.

**Comment by DaleKramer** [comment]
Sorry to bother again.

I chose to uninstall and reinstall blueCFD-Core-2017-2 in the directory 'c:\blueCFD-Core-2017-2' (not in 'Program Files' dir).

When I first started a blueCFD terminal window after this reinstall I got these errors:

```
mkdir: cannot change permissions of '/dev/shm': Permission denied
mkdir: cannot change permissions of '/dev/mqueue': Permission denied
'C:\Windows\system32\drivers\etc\hosts' -> '/etc/hosts'
'C:\Windows\system32\drivers\etc\protocol' -> '/etc/protocols'
'C:\Windows\system32\drivers\etc\services' -> '/etc/services'
'C:\Windows\system32\drivers\etc\networks' -> '/etc/networks'
Setting environment for OpenFOAM 5.x mingw-w64 Double Precision (of5-64), using MSMPI71 - please wait...
Environment is now ready. Notes:
   - You can change between installed versions by running: ofmenu
   - You can change to other predefined versions by running: ofmenuNew

kci@DESKTOP-3FIT2JC MINGW64 OpenFOAM-5.x ~

```

Is this another permissions issue that I should be concerned about ❓ 

Could this be the possible reason that once I install 'pacman -S base-devel' and run 'autoreconf -fi' in my MOAB dir (which I put in my user dir), that there is no 'configure' file created in the MOAB dir ❓

Also, I find it strange that when I ran the  'autoreconf -fi' and other autoreconf commands, there was no error but also no response from any of them:
![image](https://user-images.githubusercontent.com/7749699/74436993-600e3b00-4e35-11ea-9027-b4acf3f41301.png)

**Comment by wyldckat** [comment]
The first error about permissions can be ignored, namely these lines can be ignored:
```
mkdir: cannot change permissions of '/dev/shm': Permission denied
mkdir: cannot change permissions of '/dev/mqueue': Permission denied
```

As for building MOAB, I tested the following with success in an up-to-date installation of MSys2 - https://www.msys2.org/ - so after I had a new MSys2 installed, I ran these commands:

```
cd
pacman -S base-devel git nano mingw-w64-x86_64-toolchain
git clone https://bitbucket.org/fathomteam/moab.git
autoreconf -fi
./configure
```

I didn't continue, since it was taking so long to generate the configuration files, so I expected it would take a lot of time to compile.

The MSys2 installation that comes with blueCFD-Core 2017-1 is outdated, so you will likely have to update your MSys2 installation or do a new separate installation, as I mentioned above.

**Comment by DaleKramer** [comment]
WOW, that is great, a friend just got MOAB running on Ubuntu 18.04 but I would prefer blueCFD...

You say 'The MSys2 installation that comes with blueCFD-Core 2017-1 is outdated'.... 

Is blueCFD-Core 2017-2, the one I installed, outdated too, it must be as I had those errors in 2017-2 ❓ 

So, I will try to install that latest MSys2 from the links you have provided, thanks....

**Comment by wyldckat** [comment]
Sorry, I had in the back of my mind wanting to mention blueCFD-Core 2020-1 that is planned but not yet in development.
I meant to write that blueCFD-Core 2017-2 has a fairly outdated installation of MSys2, since it's from 2018: http://bluecfd.github.io/Core/ReleaseNotes/bluecfd-core-2017-2/#software-provided-with-installers

>  MSys2 x86_64, from 2018-02-20.
